Sources/MessagingPush/RichPush/RichPushRequest.swift

```diff
@@ -29,7 +26,7 @@ class RichPushRequest {
             guard let self = self else { return }
 
             if let localFilePath = localFilePath {
-                self.pushContent.addImage(localFilePath: localFilePath)
+                self.push.cioRichPushImageFile = localFilePath
```

Comment:
directly mutating push within the same class can lead to issues, especially in a multi-threaded environment as `start()` could potentially be called from different threads.

should we ensure modifications to push are thread-safe.?

Reply:
Glad you are asking questions about threading. Always important to ask. 

This PR is meant to be a full refactor, where logic is not modified. So with this line of code, the original implementation was also performing a mutation. 

Since this code is stable in production today, and my understanding that this class will not be called by multiple threads in a NSE target, I do not think we need to consider this use case in this class.

Reply:
Good points, unfortunately didn't review or missed in initial implementation. 

Following things are in my mind
- Previously `self.push` alternative was a `immutable` its mutable now.

> class will not be called by multiple threads in a NSE target
I am not sure how certain can we be on this? because i haven't found any documentation on it. Even this is true we still need to make sure we are catering for multi threading, while making http requests that might run on background thread while NSE execution might be being done on main. 

> This PR is meant to be a full refactor, where logic is not modified. So with this line of code, the original implementation was also performing a mutation.
I am okay, if this is not taken care of in this PR, that would make sense. But can we cater in another PR and have ticket for it?